### ISSUE TITLE
How can I move rocket chat to other server?

### ISSUE THREAD

**Issue by karamata** [RocketChat/Rocket.Chat#4297]
Your Rocket.Chat version: Rocket.Chat-0.39.0.zip

Dear all,
I have two question:
1. I have deploy rocket chat to old server, and now I want to move all RC app to other server. how can I do that? do I need export mongodb and import to other server
2. I use https://gonative.io/ to generate mobile app for ios and android for my rocket web app, I have modify source code to implement push and receive notification on web and mobile. but I don't know how to do. do anyone can help me?

**Comment by geekgonecrazy** [comment]
@karamata do a mongo export:
```
mongodump -o ~/backup
```

tar it up
```
tar cvf backup.tar ~/backup
```

copy it over to new host.

``` 
tar xvf backup.tar
```

Then do a mongo restore
```
mongorestore ~/backup
```

Please refer to mongo's documentation for [mongodump](https://docs.mongodb.com/manual/reference/program/mongodump/) and [mongorestore](https://docs.mongodb.com/manual/reference/program/mongorestore/)

**Comment by birchcode** [comment]
This is almost a seamless process, however I encountered an issue following those mongodb backup steps.

The mongodump, and restore seemed to work as expected, however on the new server we got rooms somehow mixed up. For example, our private team channel is nowhere to be found with the error "No private group with name "[privatename]" was found", meanwhile our #general channel has all the messages from the private channel. 

What has happened?

**Comment by birchcode** [comment]
Additionally, if we now try to create the channel again using the previous private channel name, we get 'internal server error'. I'm guessing some channel id's have been crossed in the process.

**Comment by birchcode** [comment]
To fix this myself, I did the following. This is for centos6

1. Stop mongodb  both new and old server.
2. Made a copy/tar of the whole database from old server, e.g. 
3. Copied to new server, extracted to mongodb data dir and started up mongod and restarted rocket.chat

Check permissions to make sure they are setup right on the new server. Owner and Group look like they should be set to mongod

**Comment by nipunamsi** [comment]
Hi Guys,
I'm trying to move ubuntu 16 to 14 and i getting the below issue.

/lib/x86_64-linux-gnu/libcrypto.so.1.0.0: version `OPENSSL_1.0.2' not found

anyone can please help me out. :(

**Comment by geekgonecrazy** [comment]
https://rocket.chat/docs/getting-support/#getting-support

**Comment by fuzunspm** [comment]
i'm getting
`command mongodump not found`

installed via snap on ubuntu

**Comment by geekgonecrazy** [comment]
check FAQ for backupdb uses the mongodump built into the snap: https://rocket.chat/docs/installation/manual-installation/ubuntu/snaps/#how-do-i-backup-my-snap-data

**Comment by Hugo-Trentesaux** [comment]
> @karamata do a mongo export:
> 
> ```
> mongodump -o ~/backup
> ```

    assertion: 17369 Backing up users and roles is only supported for clusters with auth schema versions 1 or 3, found: 5

**Comment by geekgonecrazy** [comment]
this is referring to mongo level users and roles not Rocket.Chat users and roles.  For what that's worth

**Comment by SecKarma** [comment]
@geekgonecrazy I've tried asking this question many times on the support channel but no one ever answered back.
I get it and i know how to backup mongo, but when i restore it, all the private messages will just disappear.

is there a way to backup private messages before moving into a dockerized environemtn or a new server? i can't stay on an old version, and i need to update it and dockerize it while preserving private chats

**Comment by manacim** [comment]
Just found out that you have to add the --drop flag when doing mongorestore. Otherwise, you'll see nothing but duplicate key errors.

**Comment by geekgonecrazy** [comment]
This is only necessary if you are attempting to import where the same db has already been imported.  I'd be very very certain when doing that.

**Comment by manacim** [comment]
@geekgonecrazy I had to use --drop flag from old server to new server, where new server's rocket instance was running but not setup past the wizard.

**Comment by geekgonecrazy** [comment]
@manacim right important to be very cautious that doing this on the right server and that you truly want to drop the db there.  In this case makes sense cause the db is auto created by rocket.chat if doesn't exist

**Comment by manacim** [comment]
@geekgonecrazy yup. Migration was smooth. 👍

**Comment by M034B5** [comment]
yup all went good

**Comment by airdogvan** [comment]
I did the same as birchcode but without stopping mongo!
Because mongo was running in docker if I stopped mongo I also stopped the container which made it impossible to attach to it and do the untar.

So I had nothing to lose, did the untar while mongo was running, telling myself that this would be a disaster but it worked!

Tried before that the dump and restore and that only have me back the general channel, all the others were gone.  So desperate I tried and it worked.

Everything is there.

Just thought others might want to know...
